sys/arm64/include/param.h

```diff
@@ -47,8 +47,17 @@
 #ifndef MACHINE
 #define	MACHINE		"arm64"
 #endif
-#ifndef MACHINE_ARCH
-#define	MACHINE_ARCH	"aarch64"
+#if !__has_feature(capabilities)
+# ifndef MACHINE_ARCH
+#  define	MACHINE_ARCH	"aarch64"
+# endif
+#else
+# ifndef MACHINE_ARCH
```

Comment:
This will make a hybrid make(1) report `aarch64c` as its MACHINE_ARCH causing some confusion (kp@ ran into this over in stock FreeBSD when dealing with riscv64 vs riscv64sf, make doesn't use the sysctl, but instead compiles in MACHINE_ARCH at build time from this header).  You probably want to follow the model RISC-V uses here of something like:

```
#if defined(__CHERI_PURE_CAPABILITY__) || (defined(_KERNEL) && __has_feature(capabilities))
#define MACHINE_ARCH "aarch64c"
#else
#define MACHINE_ARCH "aarch64"
#endif
```

You also want to define MACHINE_ARCHES at least in the CHERI case.

Reply:
Given arm64 only has aarch64 and aarch64c with no sf variants it might get away with the COMPAT_FREEBSD64 default in kern_mib.c, though I know you want to get rid of that code, and it doesn't work correctly if you have both COMPAT_FREEBSD32 and COMPAT_FREEBSD64 defined in your kernel (but we don't support the former for CHERI-aware kernels... at least not currently).

Reply:
Ah, I thought I had axed the `MACHINE_ARCHES` case for `MACHINE_ARCH64` entirely, but I guess not yet.  I think we probably want to not worry about `COMPAT_FREEBSD` right now since it would need a lot of USER_CAP_* splatted around.  The first bit about fixing `MACHINE_ARCH` for userland still applies though.